pkg/hooks/audit_logger_sqlite.go

```diff
@@ -0,0 +1,143 @@
+package hooks
+
+import (
+	"database/sql"
+
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/square/p2/pkg/logging"
+	"github.com/square/p2/pkg/util"
+)
+
+type SQLiteAuditLogger struct {
+	sqlite *sql.DB
+	logger *logging.Logger
+}
+
+func NewSQLiteAuditLogger(sqliteDBPath string, logger *logging.Logger) (*SQLiteAuditLogger, error) {
+	db, err := sql.Open("sqlite3", sqliteDBPath)
+	if err != nil {
+		return nil, util.Errorf("Unable to open sqlite at %s. %v", sqliteDBPath, err)
+	}
+	if db == nil {
+		return nil, util.Errorf("Unable to create a sqlite connection.")
+	}
+	al := &SQLiteAuditLogger{
+		sqlite: db,
+		logger: logger,
+	}
+	if err := al.ensureMigrated(); err != nil {
+		return nil, util.Errorf("Failed to apply migrations to sqlite DB: %v", err)
+	}
+	return al, nil
+}
+
+func (al *SQLiteAuditLogger) log(ctx *HookExecContext, success bool) {
+	env := ctx.env
+	stmt := `INSERT INTO hook_results(
+pod_id,
+pod_unique_key,
+hook_name,
+hook_stage,
+success
+) VALUES(?,?,?,?,?)`
+	podID := env.HookedPodIDEnvVar
+	podUniqueKey := env.HookedPodUniqueKeyEnvVar
+	hookName := ctx.Name
+	hookStage := env.HookEventEnvVar
+	// Boolean values are represented by TINYINTs in SQL. Adapt.
+	dbSuccess := 0
+	if !success {
+		dbSuccess = 1
```

Comment:
so I guess this is going to be fine if we're consistent about it (?) but this will be in contravention of https://sqlite.org/datatype3.html#boolean_datatype - we have 1 false and 0 true. Makes sense with exit codes though.